pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -553,9 +555,8 @@ public void sendAsync(Message<?> message, SendCallback callback) {
                 synchronized (this) {
                     // Update the message metadata before computing the payload chunk size
                     // to avoid a large message cannot be split into chunks.
-                    final long sequenceId = updateMessageMetadataSequenceId(msgMetadata);
-                    String uuid = totalChunks > 1 ? String.format("%s-%d", producerName, sequenceId) : null;
-
+                    final long sequenceId = updateMessageMetadataSequenceId(msgMetadata, chunkId > 0);
+                    String uuid = totalChunks > 1 ? String.format("%s-%s", producerName,  sequenceId - chunkId) : null;
```

Comment:
Shouldn't chunk uuid be the same for all chunks per message?

Reply:
The sequence ID will increase with the chunk ID. So they have the same uuid for all chunks per message.